core/includes/install.core.inc

```diff
@@ -1307,7 +1280,7 @@ function install_find_translation_files($langcode = NULL) {
   $directory = settings_get('locale_translate_file_directory', conf_path() . '/files/translations');
   // @todo: Remove the "drupal" check in 2.x (assuming we have a localization
   //   server by then).
-  $files = file_scan_directory($directory, '!(install|drupal|backdrop(cms)?)(-[\d\.]+)?\.' . (!empty($langcode) ? preg_quote($langcode, '!') : '[^\.]+') . '\.po$!', array('recurse' => FALSE));
+  $files = file_scan_directory($directory, '!(install|drupal|backdrop(cms)?)(-[\d]+\.)?([\d]+|x)?\.' . (!empty($langcode) ? preg_quote($langcode, '!') : '[^\.]+') . '\.po$!', array('recurse' => FALSE));
```

Comment:
@docwilmot I've already raised https://github.com/backdrop/backdrop-issues/issues/6662 in order to clean up the regex here separately without getting out of scope here, but can you please explain what we are trying to catch and what wasn't working before? Reason I'm asking is because this change here feels overly complex for what I think it is trying to achieve. Were you trying to catch any `.x` in the version number?

Reply:
We accept:
install.ca.po (used in tests)
backdropcms-1.28.ca.po 
backdrop-1.x.ca.po
drupal-7.x.ca.po
backdropcms-1.x.ca.po (this is the reason for the PR change. If this is the http request, the localize site serves the latest langauge file but named as 1,x)

Reply:
Thanks @docwilmot 🙏🏼 ...this can then be simplified to the following:
```suggestion
  $files = file_scan_directory($directory, '!(install|drupal|backdrop(cms)?)(-[\d\.x]+)?\.' . (!empty($langcode) ? preg_quote($langcode, '!') : '[^\.]+') . '\.po$!', array('recurse' => FALSE));
```

PS: I'm not sure what the `'[^\.]+'` is meant to catch there, since `^` signifies the beginning of a string in a regex. But that's out of scope to discuss here.

Reply:
Done